drivers/serial/serial_cmsdk.c

```diff
@@ -0,0 +1,796 @@
+
+  up_disable_irq(priv->tx_irq);
+  irq_detach(priv->tx_irq);
+  up_disable_irq(priv->rx_irq);
+  irq_detach(priv->rx_irq);
+  up_disable_irq(priv->ov_irq);
+  irq_detach(priv->ov_irq);
+}
+
+/****************************************************************************
+ * Name: uart_cmsdk_interrupt
+ *
+ * Description:
+ *   This is the UART interrupt handler.  It will be invoked when an
+ *   interrupt received on the 'irq'  It should call uart_transmitchars or
+ *   uart_receivechar to perform the appropriate data transfers.  The
+ *   interrupt handling logic must be able to map the 'irq' number into the
+ *   appropriate uart_cmsdk_s structure in order to call these functions.
+ *
+ ****************************************************************************/
+
+static int uart_cmsdk_rx_interrupt(int irq, FAR void *context, FAR void *arg)
+{
+  FAR struct uart_dev_s *dev = (struct uart_dev_s *)arg;
+  FAR struct uart_cmsdk_s *priv;
+
+  DEBUGASSERT(dev != NULL && dev->priv != NULL);
+  priv = (FAR struct uart_cmsdk_s *)dev->priv;
+
+  uart_cmsdk_serialout(priv, UART_INTSTS_OFFSET, UART_INTSTATUS_RX);
+  uart_recvchars(dev);
+
+  return OK;
+}
+
+static int uart_cmsdk_ov_interrupt(int irq, FAR void *context, FAR void *arg)
+{
+  FAR struct uart_dev_s *dev = (struct uart_dev_s *)arg;
+  FAR struct uart_cmsdk_s *priv;
+
+  DEBUGASSERT(dev != NULL && dev->priv != NULL);
+  priv = (FAR struct uart_cmsdk_s *)dev->priv;
+  if (uart_cmsdk_serialin(priv, UART_INTSTS_OFFSET) &
+     UART_INTSTATUS_RX_OVERRUN)
+    {
+      uart_cmsdk_serialout(priv, UART_INTSTS_OFFSET,
+                           UART_INTSTATUS_RX_OVERRUN);
+      uart_cmsdk_serialout(priv, UART_STATE_OFFSET,
+                           UART_STATE_RX_BUF_OVERRUN);
+    }
+
+  if (uart_cmsdk_serialin(priv, UART_INTSTS_OFFSET) &
+     UART_INTSTATUS_TX_OVERRUN)
+    {
+      uart_cmsdk_serialout(priv, UART_INTSTS_OFFSET,
+                           UART_INTSTATUS_TX_OVERRUN);
+      uart_cmsdk_serialout(priv, UART_STATE_OFFSET,
+                           UART_STATE_TX_BUF_OVERRUN);
+    }
+
+  return OK;
+}
+
+static int uart_cmsdk_tx_interrupt(int irq, FAR void *context, FAR void *arg)
+{
+  FAR struct uart_dev_s *dev = (struct uart_dev_s *)arg;
+  FAR struct uart_cmsdk_s *priv;
+
+  DEBUGASSERT(dev != NULL && dev->priv != NULL);
+  priv = (FAR struct uart_cmsdk_s *)dev->priv;
+
+  uart_cmsdk_serialout(priv, UART_INTSTS_OFFSET, UART_INTSTATUS_TX);
+  uart_xmitchars(dev);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: uart_cmsdk_ioctl
+ *
+ * Description:
+ *   All ioctl calls will be routed through this method
+ *
+ ****************************************************************************/
+
+static int uart_cmsdk_ioctl(struct file *filep, int cmd, unsigned long arg)
+{
+  FAR struct uart_dev_s *dev = filep->f_inode->i_private;
+  FAR struct uart_cmsdk_s *priv = (FAR struct uart_cmsdk_s *)dev->priv;
```

Comment:
```suggestion
  FAR struct uart_cmsdk_s *priv = dev->priv;
```
remove all `void *` cast